android/app/src/main/java/com/microsoft/codepush/react/CodePushNativeModule.java

```diff
@@ -240,19 +245,20 @@ public void dispatchDownloadProgressEvent() {
 
     @ReactMethod
     public void getConfiguration(Promise promise) {
-        Activity currentActivity = getCurrentActivity();
         WritableNativeMap configMap = new WritableNativeMap();
         configMap.putString("appVersion", mCodePush.getAppVersion());
+        configMap.putString("clientUniqueId", mClientUniqueId);
         configMap.putString("deploymentKey", mCodePush.getDeploymentKey());
         configMap.putString("serverUrl", mCodePush.getServerUrl());
-        configMap.putString("clientUniqueId",
-                Settings.Secure.getString(currentActivity.getContentResolver(),
-                        android.provider.Settings.Secure.ANDROID_ID));
-        String binaryHash = CodePushUpdateUtils.getHashForBinaryContents(currentActivity, mCodePush.isDebugMode());
-        if (binaryHash != null) {
-            // binaryHash will be null if the React Native assets were not bundled into the APK
-            // (e.g. in Debug builds)
-            configMap.putString(CodePushConstants.PACKAGE_HASH_KEY, binaryHash);
+
+        Activity currentActivity = getCurrentActivity();
+        if (currentActivity != null) {
+            String binaryHash = CodePushUpdateUtils.getHashForBinaryContents(currentActivity, mCodePush.isDebugMode());
```

Comment:
OK cool, I'll move this to app start and update the until method to take a Context instead of an Activity. Then our only dependency on getCurrentActivity will be the restart logic, which I don't see a way around unfortunately.